src/engine/source/metrics/src/metrics.cpp

```diff
@@ -98,6 +104,7 @@ void Metrics::createCommonChain(const std::filesystem::path& file)
 void Metrics::setMetricsConfig()
 {
     auto particularContext = m_upContext.begin();
+    m_instrumentState.clear();
```

Comment:
Why is it necessary to clean only this map?

Reply:
this is because the map is initialized with default values so that the compiler does not complain about the empty constructor.